prow/plugins/projectmanager/projectmanager.go

```diff
@@ -0,0 +1,204 @@
+limitations under the License.
+*/
+
+// Package projectmanager is a plugin to auto add pull requests to project boards based on specified conditions
+package projectmanager
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	pluginName = "project-manager"
+)
+
+// TODO Create a new handler for issues, look in hook/server.go
+func init() {
+	plugins.RegisterPullRequestHandler(pluginName, handlePullRequest, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The project-manager plugin automatically adds Pull Requests to specified GitHub Project Columns if they match given criteria.",
+		Config: func(config *plugins.Configuration) map[string]string {
+			configMap := make(map[string]string)
+			configString := "org/repos: {"
+			for orgRepoName, managedOrgRepo := range config.ProjectManager.OrgRepos {
+				configString := fmt.Sprintf("%s %s: { projects: {", configString, orgRepoName)
+				for projectName, managedProject := range managedOrgRepo.Projects {
+					configString := fmt.Sprintf("%s %s: { columns: [", configString, projectName)
+					for _, managedColumn := range managedProject.Columns {
+						configString := fmt.Sprintf("%s {id: \"%d\", name: \"%s\", state: \"%s\", org: \"%s\" labels: [", configString, managedColumn.ID, managedColumn.Name, managedColumn.State, managedColumn.Org)
+						for i, label := range managedColumn.Labels {
+							configString = fmt.Sprintf("%s \"%s\"", configString, label)
+							if i+1 < len(managedColumn.Labels) {
+								configString = fmt.Sprintf("%s,", configString)
+							}
+						}
+						configString = fmt.Sprintf("%s ] }", configString)
+					}
+					configString = fmt.Sprintf("%s ] }", configString)
+				}
+				configString = fmt.Sprintf("%s }", configString)
+			}
+			configString = fmt.Sprintf("%s }", configString)
+			configMap[""] = configString
+			return configMap
+		}(config),
+	}
+	return pluginHelp, nil
+}
+
+func handlePullRequest(pc plugins.Agent, pe github.PullRequestEvent) error {
+	return handlePR(pc.GitHubClient, pc.PluginConfig.ProjectManager, pc.Logger, pe)
+}
+
+// Strict subset of *github.Client methods.
+type githubClient interface {
+	GetIssueLabels(org, repo string, number int) ([]github.Label, error)
+	GetRepoProjects(owner, repo string) ([]github.Project, error)
+	GetOrgProjects(org string) ([]github.Project, error)
+	GetProjectColumns(projectID int) ([]github.ProjectColumn, error)
+	CreateProjectCard(columnID int, projectCard github.ProjectCard) (*github.ProjectCard, error)
+}
+
+func handlePR(gc githubClient, projectManager plugins.ProjectManager, log *logrus.Entry, pe github.PullRequestEvent) error {
+	// Only respond to label add or issue/PR open events
```

Comment:
Wait, right now this is not checking for any comment command -- are we trying to add every single opened PR and issue to a board?

Reply:
If a project is configured and a label is added to the PR, we add the PR to that project under the column mentioned or in a default column on the project.